embed/config.go

```diff
@@ -363,6 +363,8 @@ type Config struct {
 	// UnsafeNoFsync disables all uses of fsync.
 	// Setting this is unsafe and will cause data loss.
 	UnsafeNoFsync bool `json:"unsafe-no-fsync"`
+	// AcceptNextStorageVersion allows server to restore from 3.5 data, to allow downgrade from 3.5 to 3.4.
+	AcceptNextStorageVersion bool `json:"accept-next-storage-version"`
```

Comment:
Users might not be aware of storage version, which was also introduced in v3.6.

Naming is hard, so let's iterate. Maybe something like "can-join-newer-cluster-version"?

Reply:
5 words is a mouthful. How about `forward-compatible`?